src/System.Runtime.Extensions/src/System/Runtime/Versioning/FrameworkName.cs

```diff
@@ -203,16 +203,17 @@ public FrameworkName(String frameworkName)
             for (int i = 1; i < components.Length; i++)
             {
                 // Get the key/value pair separated by '='
-                string[] keyValuePair = components[i].Split(c_keyValueSeparator);
+                string component = components[i];
+                int separatorIndex = component.IndexOf('=');
 
-                if (keyValuePair.Length != 2)
+                if (separatorIndex == -1 || separatorIndex != component.LastIndexOf('='))
                 {
                     throw new ArgumentException(SR.Argument_FrameworkNameInvalid, nameof(frameworkName));
                 }
 
                 // Get the key and value, trimming any whitespace
-                string key = keyValuePair[0].Trim();
-                string value = keyValuePair[1].Trim();
+                string key = component.Substring(0, separatorIndex).Trim();
+                string value = component.Substring(separatorIndex + 1).Trim();
```

Comment:
If it's not already in use in this assembly, I don't think it's worth the extra code being included.  FrameworkNames aren't instantiated very often, are they, and if they are, are they instantiated with inputs that would cause these Trims to actually trim something? (Trim already returns the original string if there's nothing to trim.)